src/main/kotlin/org/pkl/lsp/ast/PklNode.kt

```diff
@@ -824,17 +824,19 @@ fun TreeSitterNode.toNode(project: Project, parent: PklNode?): PklNode? {
         "!" -> PklLogicalNotExprImpl(project, parent!!, this)
         else -> PklNonNullExprImpl(project, parent!!, this)
       }
-    "binaryExprRightAssoc" ->
-      when (children[1].type) {
+    "binaryExprRightAssoc" -> {
+      val operator = children.find { it.type == "**" || it.type == "??" }
+      when (operator?.type) {
         "**" -> PklExponentiationExprImpl(project, parent!!, this)
         "??" -> PklNullCoalesceExprImpl(project, parent!!, this)
-        "lineComment",
-        "blockComment" -> null
-        "ERROR" -> PklErrorImpl(project, parent!!, this)
-        else -> throw RuntimeException("Unknown binary operator `${children[1].type}`")
+        else -> throw RuntimeException("Unknown binary operator in expr `$text`")
       }
-    "binaryExpr" ->
-      when (children[1].type) {
+    }
+    "binaryExpr" -> {
```

Comment:
Nit: you can combine `"binaryExprRightAssoc"` and `"binaryExpr"` into the same block, since they basically have the same logic.